### PR TITLE
Issue 3540: Fix baseUrl slash

### LINKED ISSUE #3540: Cypress.config('baseUrl') escapes slash
### Current behavior:

image

### Desired behavior:

image

### Steps to reproduce: (app code and test code)

cypress.json
```json
{
  "env": {
    "test_url": "http://localhost:3000/"
  },
  "baseUrl": "http://localhost:3000/"
}
```
Example test:
```js
it('test url', function () {
    cy.visit('/')
    var baseUrl = Cypress.config('baseUrl')
    //baseUrl = baseUrl + '/'    // bug in Cypress - add slash
    cy.url().should('eq', Cypress.env('test_url'))
    cy.url().should('eq', baseUrl)
})
```

### Versions

Windows, Cypress 3.1.5

### PR DESCRIPTION
**Pull request by Saibamen**
closes #3540 and #3545 

Tests:
```js
/// <reference types="Cypress" />

it('test url', function () {
    cy.visit('/')
    var baseUrl = Cypress.config('baseUrl')
    //baseUrl = baseUrl + '/'    // bug in Cypress - add slash
    cy.url().should('eq', Cypress.env('test_url'))
    cy.url().should('eq', baseUrl)
})

it('test url - copy paste from @jennifer-shehane', function () {
    cy.visit('/')
    cy.url().should('eq', Cypress.config('baseUrl'))
})
```

obraz

### CODE REVIEW

File: packages/server/lib/config.coffee [hunk 1]
```diff
@@ -230,8 +230,9 @@ module.exports = {
     .value()
 
     if url = config.baseUrl
-      ## always strip trailing slashes
-      config.baseUrl = _.trimEnd(url, "/")
+      ## always strip trailing slashes and add one to end
```
Comment (chrisbreiding) [inline packages/server/lib/config.coffee]: I'm not sure we should always be adding a slash anymore than we should always be removing one. Seems like the ideal is to leave the baseUrl as the user has it in their config, while also stripping extra `/`s. So `http://localhost:8080/` and `http://localhost:8080` stay the same and `http://localhost:8080///` becomes `http://localhost:8080/`.

Haven't tested this, but something like like `config.baseUrl = url.replace(/\/\\+$/, '/')` should work.

Comment (Saibamen) [reply packages/server/lib/config.coffee]: Done: https://regexr.com/48rvt

File: packages/server/lib/config.coffee [hunk 2]
```diff
@@ -230,8 +230,10 @@ module.exports = {
     .value()
 
     if url = config.baseUrl
-      ## always strip trailing slashes
-      config.baseUrl = _.trimEnd(url, "/")
+      ## replace multiple slashes at the end of string to single slash
+      ## so http://localhost/// will be http://localhost/
+      ## https://regexr.com/48rvt
+      config.baseUrl = url.replace(/\/\/+$/, "/")
```
Comment (brian-mann) [inline packages/server/lib/config.coffee]: regular expressions should be cached at the top of the file and dynamically generated... this also allows them to have a real name to provide additional context.

something like this... `multipleBackslashesRe`

also - whenever fixing bugs like this we need to add a comment at the top of the fix that points to the corresponding github issue to help developers understand **why** this change is necessary.

Comment (jennifer-shehane) [reply packages/server/lib/config.coffee]: @brian-mann will fix directly in dev. No action required here.

**Pull request merged**